src/evaluate/evaluator/text_generation.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 The HuggingFace Evaluate Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Tuple
+
+from datasets import Dataset
+
+from .base import Evaluator
+from .utils import DatasetColumn
+
+
+TASK_DOCUMENTATION_KWARGS = r"""
+        input_column (`str`, defaults to `"text"`):
+            the name of the column containing the input text in the dataset specified by `data`.
+        generation_kwargs (`Dict`, *optional*, defaults to `None`):
+            The generation kwargs are passed to the pipeline and set the text generation strategy.
+"""
+
+
+class TextGenerationEvaluator(Evaluator):
+    """
+    Text generation evaluator.
+    This Text generation evaluator can currently be loaded from [`evaluator`] using the default task name
+    `text-generation`.
+    Methods in this class assume a data format compatible with the [`TextGenerationPipeline`].
+    """
+
+    def predictions_processor(self, predictions, *args, **kwargs):
+        if isinstance(predictions[0], list):
+            return {"data": [pred[0][f"{self.predictions_prefix}_text"] for pred in predictions]}
+        elif isinstance(predictions[0], dict):
```

Comment:
It returns either a list of dicts (for single string input) or a list of lists of dicts, but I made the mistake of merging `DummyTextGenerationPipeline` and `DummyText2TextGenerationPipeline` which made me think that I needed to cover the `list of dicts` case. In reality it looks like that would never happen since the Evaluator is passed an iterable, and the text-generation pipeline would then always return the list of lists of dicts.

I'll fix that!